lib/system/repr_v2.nim

```diff
@@ -92,8 +95,15 @@ proc repr*(p: proc): string =
   ## repr of a proc as its address
   repr(cast[ptr pointer](unsafeAddr p)[])
 
-template repr*(x: distinct): string =
-  repr(distinctBase(typeof(x))(x))
+template repr*[T: distinct|range](x: T): string =
+  # note: distinct and range match equally well, so it's not possible to use
+  # two templates -- doing so would lead to ambiguity errors
```

Comment:
Maybe this should become a generics known issue, `range` is more precise than `distinct` at the very least. Also, I'm pretty sure `distinct` should not mean `distinct|range[distinct]`... that seems like madness.

Reply:
Yep, I agree -- I'll add a `knownIssue` test.